### DISCUSSION TITLE
Creating both a Python package and Rust crate

### DISCUSSION THREAD

**Discussion by CarlKCarlK** [PyO3/maturin#942] [General]
Greetings,

I'd like to offer our genomics-related library, [bed-reader](https://github.com/fastlmm/bed-reader/tree/rustybed), as a Rust crate. It is already a Python package via maturin.

Has anyone else created such a project?

When I set `crate-type = ["cdylib", "rlib"]`, Linux and Windows work, but MacOS fails with 
[= note: Undefined symbols for architecture x86_64: "_PyType_IsSubtype"...](https://github.com/fastlmm/bed-reader/runs/6577797411?check_suite_focus=true#step:9:419).

I guess I should instead [create two related crates that share a Cargo workspace and repository](https://stackoverflow.com/questions/40514793/how-can-i-use-cargo-to-build-a-library-as-both-an-rlib-and-a-dylib-but-with-diff). Is that right?

Thanks,
Carl

**Comment by messense** [comment]
See https://github.com/PyO3/pyo3/discussions/2271

**Reply by messense** [reply]
In other words, you should feature-gate the python extension module code (off by default) if you want to distribute a Python package and Rust crate together.

**Reply by CarlKCarlK** [reply]
Thanks!
